src/box/xrow.c

```diff
@@ -1463,6 +1463,42 @@ xrow_decode_error(struct xrow_header *row)
 	box_error_set(__FILE__, __LINE__, code, error);
 }
 
+int
+xrow_decode_txn(const struct xrow_header *row, struct txn_request *request)
+{
+	assert(row->type == IPROTO_BEGIN);
+	memset(request, 0, sizeof(*request));
+
+	/** Request without extra options. */
+	if (row->bodycnt == 0)
+		return 0;
+
+	const char *begin = row->body[0].iov_base;
+	const char *end = begin + row->body[0].iov_len;
+	const char *pos = begin;
+	uint32_t map_size = mp_decode_map(&pos);
```

Comment:
You should check that the body is a map, like other similar functions do.

Reply:
fixed